server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -29,16 +41,25 @@ public static class Defaults {
         public static final String NAME = IgnoredFieldMapper.NAME;
     }
 
-    public static final IgnoredFieldType FIELD_TYPE = new IgnoredFieldType();
+    public static final IgnoredFieldType FIELD_TYPE = new IgnoredFieldType(true);
+    private static final IgnoredFieldMapper INSTANCE = new IgnoredFieldMapper(FIELD_TYPE);
+
+    public static final IgnoredFieldType LEGACY_FIELD_TYPE = new IgnoredFieldType(false);
+    private static final IgnoredFieldMapper LEGACY_INSTANCE = new IgnoredFieldMapper(LEGACY_FIELD_TYPE);
 
-    private static final IgnoredFieldMapper INSTANCE = new IgnoredFieldMapper();
 
-    public static final TypeParser PARSER = new FixedTypeParser(c -> INSTANCE);
+    public static final TypeParser PARSER = new FixedTypeParser(c -> getInstance(c.indexVersionCreated()));
+
+    public static MetadataFieldMapper getInstance(IndexVersion indexVersion) {
+        return indexVersion.onOrAfter(AGGS_SUPPORT_VERSION)
+            ? INSTANCE
+            : LEGACY_INSTANCE;
+    }
 
     public static final class IgnoredFieldType extends StringFieldType {
 
-        private IgnoredFieldType() {
-            super(NAME, true, true, false, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+        private IgnoredFieldType(boolean hasDocValues) {
+            super(NAME, true, true, hasDocValues, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
         }
 
```

Comment:
I wish I could  comment on parts of this file that were not modified...

You'll probably want to delete the `existsQuery` method so that it can do a faster exists query that leverages doc values.

Reply:
Done. The experts review should validate that as well.